docs/resources/server/advancements.md

```diff
@@ -92,48 +89,42 @@ public boolean matches(ItemStack stack) {
 
 ### SimpleCriterionTrigger
 
-The `SimpleCriterionTrigger<T>` subclass, where `T` is the type of the trigger instance, is responsible for specifying the registry name of the trigger, creating a trigger instance, and a method to check trigger instances and run attached listeners on success.
-
-The registry name of the trigger is supplied to `#getId`. This should match the registry name supplied to the trigger instance.
+The `SimpleCriterionTrigger<T>` subclass, where `T` is the type of the trigger instance, is responsible for specifying a codec to serialize `T`s, and supplying a method to check trigger instances and run attached listeners on success.
 
-A trigger instance is created via `#createInstance`. This method reads a criteria from JSON.
+The codec is typically defined as a constant of the inner `SimpleCriterionTrigger.SimpleInstance` class, and simply returned by the trigger's `codec()` method.
 
 ```java
+// in ExampleTrigger
 @Override
-public ExampleTriggerInstance createInstance(JsonObject json, ContextAwarePredicate player, DeserializationContext context) {
-  // Read conditions from JSON: item
-  return new ExampleTriggerInstance(player, item);
+public Codec<ExampleTriggerInstance> codec() {
+  return ExampleTriggerInstance.CODEC;
 }
```

Comment:
It would be better to combine all of the codec code blocks together.

Reply:
Gave that a go. It feels a bit awkward to retroactively describe things which go in both classes, but I agree it's better than the duplication